src/main/java/com/example/onlinebookstore/repository/book/BookRepository.java

```diff
@@ -1,10 +1,26 @@
 package com.example.onlinebookstore.repository.book;
 
 import com.example.onlinebookstore.model.Book;
+import java.util.List;
 import java.util.Optional;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.transaction.annotation.Transactional;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    @Query(value = "SELECT * FROM books b INNER JOIN books_categories bc ON "
+            + "b.id = bc.book_id WHERE bc.category_id = :categoryId", nativeQuery = true)
+    List<Book> findAllByCategoryIds(Long categoryId, Pageable pageable);
+
     Optional<Book> findBookById(Long id);
+
+    @Modifying
+    @Transactional
+    @Query(value = "DELETE FROM books_categories WHERE book_id = :bookId",
+            nativeQuery = true)
+    void removeBooksByBookId(Long bookId);
```

Comment:
Same here,
you shouldn't delete data from join table, all get queries should be automatically filtered from deleted entities

Please remove methods, usage of them and recheck if workflow for other operations is correct. If not - please raise question in chat or on Q&A, but removing data while using soft-delete policy is not a good way to go